util/reggen/gen_cheader.py

```diff
@@ -188,9 +188,50 @@ def gen_cdefines_module_params(outstr, module_data, module_name,
     genout(outstr, '\n')
 
 
-# Must have called validate, so should have no errors
+def gen_cdefines_interrupt_field(outstr, interrupt, component, regwidth, existing_defines):
+    fieldlsb = interrupt['bitinfo'][2]
+    iname = interrupt['name']
+    defname = as_define(component + '_INTR_COMMON_' + iname)
+
+    if interrupt['bitinfo'][1] == 1:
+        # single bit
+        genout(outstr,
+                gen_define(defname, [], str(fieldlsb), existing_defines))
+    else:
+        # multiple bits (unless it is the whole register)
+        if interrupt['bitinfo'][1] != regwidth:
+            mask = interrupt['bitinfo'][0] >> fieldlsb
+            genout(
+                outstr,
+                gen_define(defname + '_MASK', [], hex(mask),
+                            existing_defines))
+            genout(
+                outstr,
+                gen_define(defname + '_OFFSET', [], str(fieldlsb),
+                            existing_defines))
+
+def gen_cdefines_interrupts(outstr, regs, component, regwidth, existing_defines):
+    no_auto_i = False
+    if 'no_auto_intr_regs' in regs:
+        no_auto_intr_regs_val = regs['no_auto_intr_regs']
+        if isinstance(no_auto_intr_regs_val, bool):
+            no_auto_i = no_auto_intr_regs_val
+        elif no_auto_intr_regs_val.lower() in ["true", "false"]:
```

Comment:
What is this check for?

Reply:
"no_auto_intr_regs" is a key that controls whether interrupt registers should be automatically generated from the "interrupt_list" key. It can be `True`, `False`, `'true'` or `'false'` (the latter two are strings), and does not get normalised during validation. 

If "no_auto_intr_regs" is `True` (or `'true'`), then there are custom `INTR_STATE`, `INTR_ENABLE` and `INTR_TEST` registers which don't necessarily have the same semantics or layout as the standard interrupt registers, so I felt it safest to disable generating the macros in that case.

Reply:
That makes sense, thanks. Could you please add your explanation as a comment, as I think that is the tricky part of this change that needs explanation.

Reply:
Would it be clearer if I updated the validator to turn things that are boolean-like into booleans?

Reply:
I think it probably would, but I think this block of code still deserves a comment (important bit: what `no_auto_intr_regs` mean, and your rationale).

Reply:
I've expanded the comments here. I hope they clarify what is going on.